src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -35,13 +36,95 @@ impl PrivilegeAccess {
     pub fn create(ctx: Arc<QueryContext>) -> Box<dyn AccessChecker> {
         Box::new(PrivilegeAccess { ctx })
     }
+
+    // PrivilegeAccess checks the privilege by names, we'd need to convert the GrantObject to
+    // GrantObjectByID to check the privilege.
+    // Currently we only checks ownerships by id, and other privileges by database/table names.
+    // This will change, all the privileges will be checked by id.
+    async fn convert_grant_object_by_id(
+        &self,
+        object: &GrantObject,
+    ) -> Result<Option<GrantObjectByID>> {
+        let tenant = self.ctx.get_tenant();
+
+        let object = match object {
+            GrantObject::Database(catalog_name, db_name) => {
+                if db_name.to_lowercase() == "system" {
+                    return Ok(None);
+                }
+                let db_id = self
+                    .ctx
+                    .get_catalog(catalog_name)
+                    .await?
+                    .get_database(&tenant, db_name)
+                    .await?
+                    .get_db_info()
+                    .ident
+                    .db_id;
+                GrantObjectByID::Database {
+                    catalog_name: catalog_name.clone(),
+                    db_id,
+                }
+            }
+            GrantObject::Table(catalog_name, db_name, table_name) => {
+                if db_name.to_lowercase() == "system" {
+                    return Ok(None);
+                }
+                let catalog = self.ctx.get_catalog(catalog_name).await?;
+                let db_id = catalog
+                    .get_database(&tenant, db_name)
+                    .await?
+                    .get_db_info()
+                    .ident
+                    .db_id;
+                let table = catalog.get_table(&tenant, db_name, table_name).await?;
+                let table_id = table.get_id();
+                GrantObjectByID::Table {
+                    catalog_name: catalog_name.clone(),
+                    db_id,
+                    table_id,
+                }
+            }
+            _ => return Ok(None),
+        };
+
+        Ok(Some(object))
+    }
+
+    async fn validate_access(
+        &self,
+        object: &GrantObject,
+        privileges: Vec<UserPrivilegeType>,
+        verify_ownership: bool,
+    ) -> Result<()> {
+        let session = self.ctx.get_current_session();
+        if verify_ownership {
+            let object_by_id = self.convert_grant_object_by_id(object).await.or_else(|e| {
+                if e.code() == ErrorCode::UNKNOWN_DATABASE
+                    || e.code() == ErrorCode::UNKNOWN_TABLE
+                    || e.code() == ErrorCode::UNKNOWN_CATALOG
+                {
+                    Ok(None)
+                } else {
+                    Err(e.add_message("error on validating access"))
+                }
+            })?;
```

Comment:
```suggestion
        let object_by_id = self.convert_grant_object_by_id(object).await.or_else(|e| {
            match e.code() {
                ErrorCode::UNKNOWN_DATABASE | ErrorCode::UNKNOWN_TABLE | ErrorCode::UNKNOWN_CATALOG => Ok(None),
                _ => Err(e.add_message("error on validating access")),
            }
        })?;
```

Reply:
magic, thank you for the suggestion

Reply:
I forgot to tell the GPT4 follow our code style, so the style is broken.